lib/view.js

```diff
@@ -0,0 +1,230 @@
+/** @babel */
+
+import {Emitter} from 'event-kit';
+
+import {Path} from './models';
+import {cachedProperty, closest, dom} from './utils';
+
+
+export default class AdvancedOpenFileView {
+    constructor() {
+        this.emitter = new Emitter();
+        this.cursorIndex = null;
+        this._updatingPath = false;
+
+        // Element references
+        this.pathInput = this.content.querySelector('.path-input');
+        this.pathList = this.content.querySelector('.list-group');
+        this.parentDirectoryListItem = this.content.querySelector('.parent-directory');
+
+        // Initialize text editor
+        this.pathEditor = this.pathInput.getModel();
+        this.pathEditor.setPlaceholderText('/path/to/file.txt');
+        this.pathEditor.setSoftWrapped(false);
+
+        this.content.addEventListener('click', (ev) => {
+            // Keep focus on the text input and do not propagate so that the
+            // outside click handler doesn't pick up the event.
+            ev.stopPropagation();
+            this.pathInput.focus();
+        });
+        this.content.addEventListener('click', (ev) => {
+            let listItem = ev.target::closest('.list-item');
+            if (listItem !== null) {
+                this.emitter.emit('did-click-file', listItem.dataset.fileName);
+            }
+        });
+        this.content.addEventListener('click', (ev) => {
+            if (ev.target::closest('.add-project-folder') !== null) {
+                let listItem = ev.target::closest('.list-item');
+                this.emitter.emit('did-click-add-project-folder', listItem.dataset.fileName);
+            }
+        });
+    }
+
+    @cachedProperty
+    get content() {
+        return dom(`
+            <div class="advanced-open-file">
+                <p class="info-message icon icon-file-add">
+                    Enter the path for the file to open or create.
+                </p>
+                <atom-text-editor class="path-input" mini></atom-text-editor>
+                <ul class="list-group">
+                    <li class="list-item parent-directory">
+                        <span class="icon icon-file-directory">..</span>
+                    </li>
+                </ul>
+            </div>
+        `);
+    }
+
+    createPathListItem(path) {
+        let icon = path.isDirectory() ? 'icon-file-directory' : 'icon-file-text';
+        return `
+            <li class="list-item ${path.isDirectory() ? 'directory' : ''}"
+                data-file-name="${path.full}">
+                <span class="filename icon ${icon}"
+                      data-name="${path.fragment}">
+                    ${path.fragment}
```

Comment:
Do you care that this is an injection attack vector? I'm pretty sure that a file named something like `danger<script>alert('ha!')` would actually run the javascript. I can also imagine not caring about this because it is people own damn computers.